### PR TITLE
fix SiPixelDigisClustersFromSoA for DBG IBs

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all, address comment: https://github.com/cms-sw/cmssw/pull/34259#discussion_r662756961

#### PR validation:

Now it compiles with ` -DEDM_ML_DEBUG`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport needed.

### CODE REVIEW

**Comment by mmusich** [comment]
@smuzaffar for my own education, is there a way to test this with the DBG IBs?

**Comment by smuzaffar** [comment]
@mmusich although `please test with CMSSW_12_0_DBG_X` will test it but it will not use the extra `-DEDM_ML_DEBUG` flag, so there is no need to test it for DBG IB. I will fix the DBG Ibs configuration so that the extra flag becomes part of the release itself.

File: RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc [hunk 1]
```diff
@@ -120,7 +120,7 @@ void SiPixelDigisClustersFromSoA::produce(edm::StreamID, edm::Event& iEvent, con
 #ifdef EDM_ML_DEBUG
       ++totClustersFilled;
       LogDebug("SiPixelDigisClustersFromSoA")
-          << "putting in this cluster " << ic << " " << cluster.charge() << " " << cluster.pixelADC().size();
+          << "putting in this cluster " << ic << " " << acluster.charge << " " << acluster.isize;
```
Comment (VinInn) [inline RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:123]: its ok for the debug. Sorry fpr the mess

Comment (VinInn) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:123]: BUT need to be moved before the clear...

Comment (mmusich) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:123]:
```diff
diff --git a/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc b/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc
index 4c7eec701d3..634e4082ca7 100644
--- a/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc
@@ -113,14 +113,16 @@ void SiPixelDigisClustersFromSoA::produce(edm::StreamID, edm::Event& iEvent, con
                                                        << " size/charge " << acluster.isize << '/' << acluster.charge;
       // sort by row (x)
       spc.emplace_back(acluster.isize, acluster.adc, acluster.x, acluster.y, acluster.xmin, acluster.ymin, ic);
+#ifdef EDM_ML_DEBUG
+      LogDebug("SiPixelDigisClustersFromSoA")
+          << "putting in this cluster " << ic << " " << acluster.charge << " " << acluster.isize;
+#endif
       aclusters[ic].clear();
       std::push_heap(spc.begin(), spc.end(), [](SiPixelCluster const& cl1, SiPixelCluster const& cl2) {
         return cl1.minPixelRow() < cl2.minPixelRow();
       });
 #ifdef EDM_ML_DEBUG
       ++totClustersFilled;
-      LogDebug("SiPixelDigisClustersFromSoA")
-          << "putting in this cluster " << ic << " " << acluster.charge << " " << acluster.isize;
 #endif
     }
     nclus = -1;
```

like this?

Comment (VinInn) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelDigisClustersFromSoA.cc:123]: yes. Once can move ++totClustersFilled; as well.

**Comment by slava77** [comment]
now we have a duplicate in #34317
based on https://github.com/cms-sw/cmssw/pull/34317#issuecomment-873116243 it seems like that PR may be preferred over this one.

@mmusich please check and suggest which to pick.

**Comment by mmusich** [comment]
Since Vincenzo prefers the other one, let's take #34317

**Pull request closed without merging**